applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h

```diff
@@ -217,40 +217,62 @@ class MPMStrategy
                     Matrix shape_functions_values = rGeom.ShapeFunctionsValues( GeometryData::GI_GAUSS_2);
                     if (m_GeometryElement == "Triangle")
                     {
-                        if(m_NumPar == 1)
+                        switch (m_NumPar)
                         {
-                            shape_functions_values = rGeom.ShapeFunctionsValues( GeometryData::GI_GAUSS_1);
-                        }
-                        else if(m_NumPar == 3)
-                        {
-                            shape_functions_values = rGeom.ShapeFunctionsValues( GeometryData::GI_GAUSS_2);
-                        }
-                        else if(m_NumPar == 6)
-                        {
-                            shape_functions_values = rGeom.ShapeFunctionsValues( GeometryData::GI_GAUSS_4);
-                        }
-                        else if(m_NumPar == 16)
-                        {
-                            shape_functions_values = this->MP16ShapeFunctions();
+                            case 1:
+                                shape_functions_values = rGeom.ShapeFunctionsValues( GeometryData::GI_GAUSS_1);
+                                break;
+                            case 3:
+                                shape_functions_values = rGeom.ShapeFunctionsValues( GeometryData::GI_GAUSS_2);
+                                break;
+                            case 6:
+                                shape_functions_values = rGeom.ShapeFunctionsValues( GeometryData::GI_GAUSS_4);
+                                break;
+                            case 12:
+                                shape_functions_values = rGeom.ShapeFunctionsValues( GeometryData::GI_GAUSS_5);
+                                break;
+                            case 16:
+                                if (TDim==2){
+                                    shape_functions_values = this->MP16ShapeFunctions();
+                                    break;
+                                }
+                            case 33:
+                                if (TDim==2) {
+                                    shape_functions_values = this->MP33ShapeFunctions();
+                                    break;
+                                }
+                            default:
+                                std::string warning_msg = "The input number of particle: " + std::to_string(m_NumPar);
+                                warning_msg += " is not available for Triangular" + std::to_string(TDim) + "D.\n";
+                                warning_msg += "Available options are: 1, 3, 6, 12, 16 (only 2D), and 33 (only 2D).\n";
+                                warning_msg += "The default number of particle: 3 is currently assumed.";
+                                KRATOS_INFO("MPM_Strategy") << "WARNING: " << warning_msg << std::endl;
```

Comment:
if it is a warning you should use KRATOS_WARNING

Reply:
@bodhinandach  and @philbucher: as far as I understand you reached an agreement on leaving KRATOS_INFO for the readability of the output file. 
If that is the case, the commit is fine for me. We already discussed it with Nanda.